x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/dimension_panel.test.tsx

```diff
@@ -517,6 +517,77 @@ describe('IndexPatternDimensionPanel', () => {
       ).not.toContain('incompatible');
     });
 
+    it('should indicate document compatibility with selected field operation', () => {
+      const initialState: IndexPatternPrivateState = {
+        ...state,
+        layers: {
+          first: {
+            ...state.layers.first,
+            columns: {
+              ...state.layers.first.columns,
+              col2: {
+                dataType: 'number',
+                isBucketed: false,
+                label: '',
+                operationType: 'avg',
+                sourceField: 'bytes',
+              },
+            },
+          },
+        },
+      };
+      wrapper = mount(
+        <IndexPatternDimensionPanel {...defaultProps} state={initialState} columnId="col2" />
+      );
+
+      openPopover();
+
+      wrapper.find('button[data-test-subj="lns-indexPatternDimension-count"]').simulate('click');
+
+      const options = wrapper.find(EuiComboBox).prop('options');
+
+      expect(options![0].className).not.toContain('incompatible');
+      options![1].options!.map(({ className }) => expect(className).toContain('incompatible'));
```

Comment:
I think we can add arbitrary attributes to the options list- they seem to get applied directly to the DOM though, so maybe a data- attribute could work instead of a className? That being said, this test is fine, I'm just worried about the usage of classNames as testable properties.

Reply:
Makes sense, I changed that to use the `data-test-subj` attribute (e.g. `lns-indexPatternDimensionIncompatible-terms`), basically treating the compatible and the incompatible option as two different test subjects. This has the additional advantage of tests failing early if they try to click an option thinking it's compatible when it's not.